src/envoy/http/authn/authn_utils_test.cc

```diff
@@ -0,0 +1,165 @@
+/* Copyright 2018 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "src/envoy/http/authn/authn_utils.h"
+#include "common/common/base64.h"
+#include "src/envoy/http/authn/authenticator_base.h"
+#include "src/envoy/http/authn/test_utils.h"
+#include "test/test_common/utility.h"
+
+using google::protobuf::util::MessageDifferencer;
+using istio::authn::JwtPayload;
+
+namespace iaapi = istio::authentication::v1alpha1;
+
+namespace Envoy {
+namespace Http {
+namespace Istio {
+namespace AuthN {
+namespace {
+
+const std::string kSecIstioAuthUserInfoHeaderKey = "sec-istio-auth-userinfo";
+const std::string kSecIstioAuthUserinfoHeaderValue =
+    R"(
+     {
+       "iss": "foo@developer.gserviceaccount.com",
```

Comment:
Make domains and aud shorter too. Use some simple schema to make it easy to see what is what in the test expect statatement (e.g iss = "issuer@foo.com", sub = "sub@foo.com", aud = "aud1" etc)

Reply:
Done.